src/components/FollowButton/FollowButton.js

```diff
@@ -0,0 +1,121 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Button from 'react-spatial/components/Button';
+import centerActive from '../../images/FollowButton/centerActive.png';
+import centerInactive from '../../images/FollowButton/centerInactive.png';
+
+import TrackerLayer from '../../layers/TrackerLayer';
+
+const propTypes = {
+  /**
+   * CSS class of the follow button.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Title.
+   */
+  title: PropTypes.string,
+
+  /**
+   * Line info route identifer.
+   */
+  routeIdentifier: PropTypes.string.isRequired,
+
+  /**
+   * Trackerlayer.
+   */
+  trackerLayer: PropTypes.instanceOf(TrackerLayer).isRequired,
+
+  /**
+   * Function to set the map center, Used to follow a train.
+   */
+  setCenter: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  className: 'rt-control-button rt-route-follow',
+  title: 'Follow',
+};
+
+/**
+ * Button enables the follow of a selected train.
+ */
+class FollowButton extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      centerActived: false,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { routeIdentifier } = this.props;
+    if (routeIdentifier !== prevProps.routeIdentifier) {
+      this.changeRouteIdentifier();
+    }
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.updateInterval);
+  }
+
+  toggleFollow(routeIdentifier) {
+    const { centerActived } = this.state;
+    const { trackerLayer, setCenter } = this.props;
+
+    const activated = !centerActived;
+
+    if (activated && trackerLayer && trackerLayer.tracker) {
+      const [trajectory] = trackerLayer.tracker
+        .getTrajectories()
+        .filter(r => r.routeIdentifier === routeIdentifier);
+      const firstCoord = trajectory && trajectory.coordinate;
+      if (firstCoord) {
+        setCenter(firstCoord);
+        this.updateInterval = window.setInterval(() => {
+          const [clickedRoute] = trackerLayer.tracker
+            .getTrajectories()
+            .filter(r => r.routeIdentifier === routeIdentifier);
+          if (clickedRoute && clickedRoute.coordinate) {
+            setCenter(clickedRoute.coordinate);
```

Comment:
there is a duplication of code here. May be a `trackerLayer.getVehicle(filterFc)`  function makes sense here. It's probably something reusable.